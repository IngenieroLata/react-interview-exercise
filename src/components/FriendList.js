import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './FriendList.css';
import FriendListItem from './FriendListItem';

class FriendList extends Component {
  render() {
    return (
      <ul className={styles.friendList}>
        {this.props.friends.map(friend => {
          return (
            <FriendListItem
              key={friend.id}
              {...friend}
              {...this.props.actions}
            />
          );
        })}
      </ul>
    );
  }
}

FriendList.propTypes = {
  friends: PropTypes.array.isRequired,
  actions: PropTypes.object.isRequired
};

export default FriendList;
