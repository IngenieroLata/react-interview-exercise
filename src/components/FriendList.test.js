import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import FriendList from './FriendList';

const props = {
  friends: [
    {
      name: 'Theodore Roosevelt',
      starred: true,
      gender: 'male'
    },
    {
      name: 'Abraham Lincoln',
      starred: false,
      gender: 'male'
    },
    {
      name: 'George Washington',
      starred: false,
      gender: 'male'
    }
  ],
  actions: {
    starFriend: jest.fn(),
    deleteFriend: jest.fn()
  }
};

describe('<FriendList />', () => {
  it('Matches the snapshot render', () => {
    const wrapper = shallow(<FriendList {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
