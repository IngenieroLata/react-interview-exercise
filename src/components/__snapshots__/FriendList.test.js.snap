exports[`<FriendList /> Matches the snapshot render 1`] = `
<ul>
  <FriendListItem
    deleteFriend={[Function]}
    gender="male"
    id={0}
    key="0"
    name="Theodore Roosevelt"
    starFriend={[Function]}
    starred={true} />
  <FriendListItem
    deleteFriend={[Function]}
    gender="male"
    id={1}
    key="1"
    name="Abraham Lincoln"
    starFriend={[Function]}
    starred={false} />
  <FriendListItem
    deleteFriend={[Function]}
    gender="male"
    id={2}
    key="2"
    name="George Washington"
    starFriend={[Function]}
    starred={false} />
</ul>
`;
